shared/src/main/scala/effekt/symbols/Name.scala

```diff
@@ -4,26 +4,117 @@ import effekt.source.Id
 import effekt.context.Context
 
 trait Name {
```

Comment:
Should we override the equals and hash methods and delegate to `qualifiedName`?

Reply:
They are implemented as case classes that should already do what you want.

Reply:
That is: use the hash of the parents and of their fields. That's good enough, no?

Reply:
Yes that should be enough.